sdk/notificationhubs/notification-hubs/src/utils/connectionStringUtils.ts

```diff
@@ -1,7 +1,58 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { SasTokenProvider, createSasTokenProvider, parseConnectionString } from "@azure/core-amqp";
+import { SasTokenProvider, createSasTokenProvider } from "../auth/sasTokenProvider.js";
+
+/**
+ * Defines an object with possible properties defined in T.
+ */
+export type ParsedOutput<T> = { [P in keyof T]: T[P] };
+
+/**
+ * Parses the connection string and returns an object of type T.
+ *
+ * Connection strings have the following syntax:
+ *
+ * ConnectionString ::= `Part { ";" Part } [ ";" ] [ WhiteSpace ]`
+ * Part             ::= [ PartLiteral [ "=" PartLiteral ] ]
+ * PartLiteral      ::= [ WhiteSpace ] Literal [ WhiteSpace ]
+ * Literal          ::= ? any sequence of characters except ; or = or WhiteSpace ?
+ * WhiteSpace       ::= ? all whitespace characters including `\r` and `\n` ?
+ *
+ * @param connectionString - The connection string to be parsed.
+ * @returns ParsedOutput<T>.
+ */
+export function parseConnectionString<T>(connectionString: string): ParsedOutput<T> {
```

Comment:
since we're inlining this method, can we take away the generic and have it return `{
    Endpoint: string;
    SharedAccessKey?: string;
    SharedAccessKeyName?: string;
  }` ( or some named alias) directly?

I think we could then get rid of the cast if we have the output be that interface to start, and we can special case the parsing to ignore unknown key names...

Reply:
Yes, will review for next time for further optimizations